src/backend/libinput.rs

```diff
@@ -0,0 +1,329 @@
+                                let desc = self.seats.get_mut(&device_seat).expect("Recieved pointer event of non existing Seat");
+                                trace!(self.logger, "Calling on_pointer_button with {:?}", button_event.button());
+                                handler.on_pointer_button(&desc.seat, button_event.time(), match button_event.button() {
+                                    0x110 => MouseButton::Left,
+                                    0x111 => MouseButton::Right,
+                                    0x112 => MouseButton::Middle,
+                                    x => MouseButton::Other(x as u8),
+                                }, button_event.button_state().into());
+                            }
+                        }
+                    }
+                },
+                _ => {}, //FIXME: What to do with the rest.
+            }
+        };
+        Ok(())
+    }
+}
+
+impl From<::input::event::keyboard::KeyState> for ::backend::input::KeyState {
+    fn from(libinput: ::input::event::keyboard::KeyState) -> Self {
+        match libinput {
+            ::input::event::keyboard::KeyState::Pressed => ::backend::input::KeyState::Pressed,
+            ::input::event::keyboard::KeyState::Released => ::backend::input::KeyState::Released,
+        }
+    }
+}
+
+impl From<::input::event::pointer::AxisSource> for ::backend::input::AxisSource {
+    fn from(libinput: ::input::event::pointer::AxisSource) -> Self {
+        match libinput {
+            ::input::event::pointer::AxisSource::Finger => ::backend::input::AxisSource::Finger,
+            ::input::event::pointer::AxisSource::Continuous => ::backend::input::AxisSource::Continuous,
+            ::input::event::pointer::AxisSource::Wheel => ::backend::input::AxisSource::Wheel,
+            ::input::event::pointer::AxisSource::WheelTilt => ::backend::input::AxisSource::WheelTilt,
+        }
+    }
+}
+
+impl From<::input::event::pointer::ButtonState> for ::backend::input::MouseButtonState {
+    fn from(libinput: ::input::event::pointer::ButtonState) -> Self {
+        match libinput {
+            ::input::event::pointer::ButtonState::Pressed => ::backend::input::MouseButtonState::Pressed,
+            ::input::event::pointer::ButtonState::Released => ::backend::input::MouseButtonState::Released,
+        }
+    }
+}
+
+impl From<::input::event::touch::TouchEvent> for ::backend::input::TouchEvent {
+    fn from(libinput: ::input::event::touch::TouchEvent) -> Self {
+        use ::input::event::touch::{TouchEventSlot, TouchEventPosition};
+        use ::backend::TouchSlotInternal;
+
+        match libinput {
+            ::input::event::touch::TouchEvent::Down(down_event) => ::backend::input::TouchEvent::Down {
+                slot: down_event.slot().map(|x| ::backend::input::TouchSlot::new(x)),
+                x: down_event.x_transformed(/*FIXME: global.get_focused_output_for_seat(&desc.seat).width() or something like that*/ 1280),
```

Comment:
Same here (`set_output_metadata()`).

Also, we'll probably want to handle cases when a touchscreen is associated with an output forever. For example in the case of a touchscreen. Not sure how we should handle that...